File: easy/2085_count_common_words_with_one_occurrence/solution.cpp
#include <string>
#include <unordered_map>
#include <vector>

class Solution {
public:
  int countWords(std::vector<std::string> &words1,
                 std::vector<std::string> &words2) {
    std::unordered_map<std::string, int> words1Counter;
    for (const std::string &word : words1)
      ++words1Counter[word];

    std::unordered_map<std::string, int> words2Counter;
    for (const std::string &word : words2)
      ++words2Counter[word];

    int wordsCount = 0;

    for (const auto &[word, count] : words1Counter) {
      if (count == 1 && words2Counter[word] == 1)
        ++wordsCount;
    }

    return wordsCount;
  }
};

File: easy/2085_count_common_words_with_one_occurrence/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
  public int countWords(String[] words1, String[] words2) {
    final Map<String, Integer> words1Counter = new HashMap<>();
    for (final String word : words1) {
      words1Counter.put(word, words1Counter.getOrDefault(word, 0) + 1);
    }

    final Map<String, Integer> words2Counter = new HashMap<>();
    for (final String word : words2) {
      words2Counter.put(word, words2Counter.getOrDefault(word, 0) + 1);
    }

    int wordsCount = 0;

    for (final Map.Entry<String, Integer> entry : words1Counter.entrySet()) {
      if (entry.getValue() == 1 && words2Counter.getOrDefault(entry.getKey(), 0) == 1) {
        ++wordsCount;
      }
    }

    return wordsCount;
  }
}
